### DISCUSSION TITLE
How to update the input field data so formData captures it

### DISCUSSION THREAD

**Discussion by bernatfortet** [react-dropzone/react-dropzone#1367] [Q&A] [answered]
@nimaa77 solved this issue and I wanted to promote that solution since I've been looking for it for a long time.

Problem:
- react-dropzone promotes using a react state to handle the acceptedFiles.
- but if I want to use native form submission (e.g. when using Remix actions or React server actions) I want the formData to have the latest file that was accepted by dropzone.

Solution:
- @nimaa77 solved it here: https://github.com/react-dropzone/react-dropzone/issues/131#issuecomment-1423414854

Hope this helps. Ideally this would be native in react-dropzone. Maybe I'm missing that capability?

**Comment by rolandjitsu** [comment, marked as answer]
@bernatfortet thanks for referencing the issue. I'd say that https://github.com/react-dropzone/react-dropzone/issues/880#issuecomment-1728920508 is also a comprehensive example of how to achieve the same result.

I will mark this as the answer, until someone posts a different answer that's more appropriate (just to indicate that this discussion has an answer).
